source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst

```diff
@@ -0,0 +1,106 @@
+.. Copyright (C) 2019 Wazuh, Inc.
+
+.. _create-custom-wpk-automatically:
+
+Automated custom WPK packages creation
+============================
+
+Wazuh provides an automated way of building WPK packages.
+
+Requirements
+^^^^^^^^^^^^
+
+    * Docker
+
+1. Get a X509 certificate and CA
+--------------------------------
+
+Create root CA
+^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    # openssl req -x509 -new -nodes -newkey rsa:2048 -keyout wpk_root.key -out wpk_root.pem -batch
+
+Create a certificate and key
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    # openssl req -new -nodes -newkey rsa:2048 -keyout wpkcert.key -out wpkcert.csr -subj '/C=US/ST=CA/O=Wazuh'
+
+Set the location as follows:
+
+    - /C=US is the country.
+    - /ST=CA is the state.
+    - /O=Wazuh is the organization's name.
+
+Sign this certificate with the root CA
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    # openssl x509 -req -days 365 -in wpkcert.csr -CA wpk_root.pem -CAkey wpk_root.key -out wpkcert.pem -CAcreateserial
+
+2. Build WPK
+--------------------
+
+2.1. Download our wazyh-packages repository from github, there you can find tools for building diferent kinds of packages, and go to the wpk directory.
+
+  .. code-block:: console
+
+    $ git clone https://github.com/wazuh/wazuh-packages && cd wazuh-installers/wpk
+
+2.2. Execute the `generate_wpk_package.sh` script, whith the diferent options you desire. This script will build a Docker image with all the necesary tools to  build the WPK and run a container that will build it:
```

Comment:
whith?

Reply:
necesary?

Reply:
Typos corrected in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/5a5c8eee3bfe1d516cebc82e63143658667f151e